src/aggregator/aggregator/aggregator.go

```diff
@@ -223,24 +224,41 @@ func (agg *aggregator) AddUntimed(
 	union unaggregated.MetricUnion,
 	metadatas metadata.StagedMetadatas,
 ) error {
-	if agg.opts.TimedForResendEnabled() {
-		for m := range metadatas {
-			for p := range metadatas[m].Pipelines {
-				if metadatas[m].Pipelines[p].ResendEnabled {
-					if union.Type != metric.GaugeType {
-						return fmt.Errorf("cannot convert a %s to a timed metric", union.Type)
-					}
-					timedMetric := aggregated.Metric{
-						Type:       metric.GaugeType,
-						ID:         union.ID,
-						TimeNanos:  int64(union.ClientTimeNanos),
-						Value:      union.GaugeVal,
-						Annotation: union.Annotation,
-					}
-					return agg.AddTimedWithStagedMetadatas(timedMetric, metadatas)
-				}
+	if agg.opts.TimedForResendEnabled() && len(metadatas) == 1 {
```

Comment:
why do we need the len check?

Reply:
this is just easier and it's temporary until we can move the logic to the coordinator.